plugins/PromisedScene/Main.js

```diff
@@ -0,0 +1,1220 @@
+
+          let SearchedTitle = stripStr(sceneName).toString().toLowerCase();
+
+          let MatchTitle = stripStr(alltitles["Title" + idx])
+            .toString()
+            .toLowerCase();
+
+          // lets remove the actors from the scenename and the searched title -- We should already know this
+
+          for (let j = 0; j < Actor.length; j++) {
+            SearchedTitle = SearchedTitle.replace(Actor[j].toString().toLowerCase(), "");
+
+            MatchTitle = MatchTitle.replace(Actor[j].toString().toLowerCase(), "").trim();
+          }
+
+          // lets remove the Studio from the scenename and the searched title -- We should already know this
+
+          if (Studio[0] !== undefined) {
+            SearchedTitle = SearchedTitle.replace(Studio[0].toString().toLowerCase(), "");
+
+            SearchedTitle = SearchedTitle.replace(
+              Studio[0].toString().toLowerCase().replace(" ", ""),
+              ""
+            );
+
+            MatchTitle = MatchTitle.replace(Studio[0].toString().toLowerCase(), "").trim();
+          }
+
+          // Only Run a match if there is a searched title to execute a match on
+
+          if (MatchTitle !== undefined) {
+            $log(
+              `     SRCH: Trying to match title: ` +
+                MatchTitle.toString().trim() +
+                "--> " +
+                SearchedTitle.toString().trim()
+            );
+
+            MatchTitle = new RegExp(MatchTitle.toString().trim(), "i");
+
+            if (SearchedTitle !== undefined) {
+              if (SearchedTitle.toString().trim().match(MatchTitle)) {
+                correct_scene_idx = idx;
+
+                break;
+              }
+            }
+          }
+        }
+      }
+
+      // making sure the scene was found (-1 is not a proper scene value)
+      if (correct_scene_idx === -1) {
+        // Will provide a list back the user if no Scene was found
+
+        if (tpdb_scene_search_content.data.length > 1 && args.ManualTouch === true) {
+          $log(" ERR: TPDB Could NOT find correct scene info, here were the results");
+
+          return alltitles;
+        }
+
+        $log(" ERR: TPDB Could NOT find correct scene info");
+
+        const manualInfo = await ManualImport();
+        return manualInfo;
+      }
+    }
+
+    const tpdb_scene_search_data = tpdb_scene_search_content.data[correct_scene_idx];
+
+    // return all of the information to TPM
+
+    if (tpdb_scene_search_data.title !== "") {
+      // Is there a duplicate scene already in the Database with that name?
+      let FoundDupScene = false;
+      // If i decide to do anything with duplicate scenes, this variable on the next line will come into play
+      // let TheDupedScene = [];
+      if (args.SceneDuplicationCheck) {
+        $fs
+          .readFileSync(args.source_settings.Scenes, "utf8")
+          .split("\n")
+          .forEach((line) => {
+            if (line !== "") {
```

Comment:
Should use a `while` instead of `for` to avoid useless iterations
```suggestion
        const lines = $fs
          .readFileSync(args.source_settings.Scenes, "utf8")
          .split("\n");
        
        let line = lines.shift();
        while (!FoundDupScene && line) {
        	// do the checks...
        	line = lines.shift();
        }
```